src/app/rmmapi/rbwebmail.ts

```diff
@@ -412,18 +412,21 @@ export class RunboxWebmailAPI {
         return this.http.post('/rest/v1/spam/', JSON.stringify(params));
     }
 
+    // Moves to Trash if not already in Trash
+    // Deletes if currently in Trash
     public deleteMessages(messageIds: number[]): Observable<any> {
         const ids = messageIds.join(',');
         return this.http.delete(`/rest/v1/email/${ids}`);
     }
 
     public markSeen(messageId: any, seen_flag_value = 1): Observable<any> {
-        return of(null).pipe(
-            tap(() => this.messageFlagChangeSubject.next(
-                new MessageFlagChange(messageId, seen_flag_value === 1 ? true : false, null)
-            )),
-            mergeMap(() => this.http.put('/rest/v1/email/' + messageId, JSON.stringify({ seen_flag: seen_flag_value }))
-        ));
+        // return of(null).pipe(
+        //     tap(() => this.messageFlagChangeSubject.next(
+        //         new MessageFlagChange(messageId, seen_flag_value === 1 ? true : false, null)
+        //     )),
+        //   mergeMap(() =>
+            return this.http.put('/rest/v1/email/' + messageId, JSON.stringify({ seen_flag: seen_flag_value }));
```

Comment:
Note that this HTTP query will not actually execute unless the caller subscribes to it.

Reply:
Yup, like most of the message action api calls